main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -1560,11 +1560,11 @@ object Typer {
         } yield (constrs.flatten, resultTyp, resultPur, resultEff)
 
 
-      case KindedAst.Expression.NewChannel(exp, declaredType, loc) =>
+      case KindedAst.Expression.NewChannel(exp, loc) =>
         for {
           (constrs, tpe, _, eff) <- visitExp(exp)
           _ <- expectTypeM(expected = Type.Int32, actual = tpe, exp.loc)
-          resultTyp <- liftM(Type.mkChannel(declaredType, loc))
+          resultTyp <- liftM(Type.freshVar(Kind.Star, loc, text = FallbackText("channel")))
```

Comment:
Up here, instead of creating the variable, we will just pull it out of `Channel(exp, elmType, loc)` and unify with it. This will ensure that `elmType` has some inferred type later when we rebuild the AST.